declarative-examples/simple-examples/environmentNonLiteral.groovy

```diff
@@ -0,0 +1,27 @@
+pipeline {
+  /*
+   * The environment section is used for setting environment variables and will allow for
+   * expanding variable and other methods to set those values as long as the return type is a String
+   */
+  environment {
+    FOO = "BAR"
+    BUILD_NUM_ENV = currentBuild.getNumber()
+    ANOTHER_ENV = "${currentBuild.getNumber()}"
+    INHERITED_ENV = "\${BUILD_NUM_ENV} is inherited"
```

Comment:
Many people seem to have problems with quoting and escaping, so this might warrant its own explanation why it's written this way.